desktop/cmp/grouping/GroupingChooser.ts

```diff
@@ -91,21 +91,30 @@ export const [GroupingChooser, groupingChooser] = hoistCmp.withFactory<GroupingC
                         ),
                         minimal: styleButtonAsInput,
                         ...buttonProps,
-                        onClick: () => model.showEditor()
+                        onClick: () => model.toggleEditor()
                     }),
                     favoritesIcon()
                 ),
                 content: favoritesIsOpen
                     ? favoritesMenu()
-                    : editor({popoverWidth, popoverMinHeight, popoverTitle, emptyText}),
+                    : editorIsOpen
+                    ? editor({popoverWidth, popoverMinHeight, popoverTitle, emptyText})
+                    : null,
                 onInteraction: (nextOpenState, e) => {
                     if (isOpen && nextOpenState === false) {
                         // Prevent clicks with Select controls from closing popover
```

Comment:
could we perhaps DRY this up, or at least add a helper function?  (e.g. targetIsControlButtonOrPortal()).  Render method is just becoming very unwieldy